### DISCUSSION TITLE
JSON only returns a line

### DISCUSSION THREAD

**Discussion by MYETRO** [Budibase/budibase#5379] [Help]
this
https://api.cartolafc.globo.com/partidas/

it doesnt return as table with rows. can you help?

**Comment by melohagan** [comment]
Hi @MYETRO 

This is for a REST query? You will need to use the transformer to to flatten the structure: https://docs.budibase.com/docs/transformers

So roughly the structure is:
one clubes -> many id objects

You'll want to use JavaScript map function (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map) to flatten.

What makes this tricky however is that you have the id as a field: `{"clubes":{"1371":{"id":1371`

In my opinion the API itself needs to flatten this into something like `{"clubes: [ {"id":1371`

**Reply by MYETRO** [reply]
Thanks. will try. Just learning things, so it's hard for me.

their api are all like this

another example
https://api.cartola.globo.com/atletas/pontuados

the whole list
https://github.com/giovanigenerali/cartrolandofc/blob/master/nova-api.md

this one works ok
https://api.cartola.globo.com/mercado/destaques

**Reply by MYETRO** [reply]
@melohagan i go the atletas thorugh by using `return data.atletas` in the transformer, but the clubes, posicoes, status wont come through

now i have another question, atletas have a URL for their photos (foto) but i need to change the URL end from _FORMATO.png to 140x140.png

"foto": "https://s.glbimg.com/es/sde/f/2022/04/02/97f5d4131e53423d654abddad3cbc9ee_FORMATO.png"

how can i do this?
